### ISSUE TITLE
Suggestion: moving foliage?

### ISSUE THREAD

**Issue by Golblahaj** [wVibzz/ImmersiveWinds#27]
Amazing works so far on this mod, looks great! It would be great to see things like grass/wheat/ferns to move/deform slightly in the direction of the wind, would sell the immersion so much to have the plants around you be affected. Maybe even falling leaves in the forest, or leaf blocks also moving slightly? Always wanted to see one of those massive yellow-grass fields rustling in the wind, but could never find a good mod to make it work or a shader that wouldnt lag massively. Maybe it could also affect kelp, if currents ever get added in the future, would make it look so much more like an actual kelp forest!. Anyway, looking forward to the wind wisp particles, good luck with the mod and great job!

**Comment by wVibzz** [comment]
I read the start of this comment and was already thinking that shaders make plants sway already. And then you said "in the direction of the wind" and I realized. I was like wow your totally right. That would be amazing. Why have I not done this?

**Comment by wVibzz** [comment]
Also Im pretty sure my mod is compatible with [Falling Leaves](https://modrinth.com/mod/fallingleaves) so you can add that btw.

**Comment by Golblahaj** [comment]
thanks so much! looking forward to the next update :)

**Comment by wVibzz** [comment]
This will not be included in the 0.8 Update, its far to complicated for me to figure out currently. Will be looked at for future versions tho.

**Comment by exocyt0sis** [comment]
Great idea - and even more great if it'll work with the excellent mod Dynamic Trees! 👍

**Comment by wVibzz** [comment]
There shouldnt be any issue running Immersive winds and Dynamic Trees!

**Comment by exocyt0sis** [comment]
> There shouldnt be any issue running Immersive winds and Dynamic Trees!

Thanks for your reply! I was actually thinking about swaying foliage - if that idea comes into fruition for standard Minecraft trees, I hope that a future version of Immersive Winds will also support moving foliage for Dynamic Trees; they are quite more elaborate than Mojang's trees, so I'm afraid it'll take some extra work to make Immersive Winds fully interoperable.

**Comment by wVibzz** [comment]
ive gotten lanterns to be swaying, i might add other blocks, and i might experiments with swaying foliage but it might be to laggy without using a shader of some kind, but finding a shader that works in vanilla and with other popular shaders whilst also working on 19 versions is not really an option currently

**Comment by exocyt0sis** [comment]
> ive gotten lanterns to be swaying, i might add other blocks

Great progress! I totally understand if doing this without a shader will potentially introduce a ton of lag, so a compromise sounds good. Here's a few suggestions for things affected by the wind:

1. Smoke particles
2. Falling leaves (would look sick during storms!)
3. Support for [Simple Clouds](https://github.com/nonamecrackers2/simple-clouds) volumetric clouds would be really nice (and they seem to be open for an API-based solution, too - please consider getting in touch with them)
4. Perhaps sound can travel farther in the direction of the wind, too?
5. Create's windmill blocks - if the wind hits them in an optimal direction, the energy output till raise by a certain factor

**Comment by wVibzz** [comment]
1. i already have the wind wisps
2. falling leaves is already a mod and is already in vanilla minecraft aswell
3. simple clouds is forge not fabric :(
4. maybe next update? might experiment
5. this is strictly a qol mod not a functional mod, although if other mods wanted to compat with me to include things like this i would be all for it.

**Comment by exocyt0sis** [comment]
It would be really cool if you can provide an official API for Immersive Winds - then other mod creators could support you by using the API. Forge and NeoForge users would be welcome onboard using [Sinytra Connector](https://github.com/Sinytra/Connector) 👍.

**Comment by wVibzz** [comment]
yeah thats the plan, im pretty sure sinytra already works aswell. although i dont have much feedback on it

**Comment by wVibzz** [comment]
https://www.curseforge.com/minecraft/mc-mods/immersive-winds

with update 0.9 all versions will be uploaded, right now only 1 version is there

**Comment by wVibzz** [comment]
swaying grass will be a thing in the next update
